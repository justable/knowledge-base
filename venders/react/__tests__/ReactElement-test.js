import React from '../../react'
import { create as createTestRenderer } from 'react-test-renderer'

describe('ReactElement', () => {
  describe('createElement', () => {
    it('快照element的数据结构', () => {
      const element = React.createElement(
        'div',
        { key: 'div_1', className: 'container' },
        'hello world'
      )
      // toMatchSnapshot 会自动将对象转成jsx
      expect(element).toMatchSnapshot()
    })
  })
  describe('cloneElement', () => {
    it('是浅克隆', () => {
      const style = { display: 'block' }
      const oldElement = React.createElement('div', { style }, 'hello world')
      expect(React.isValidElement(oldElement)).toBeTruthy()
      const newElement = React.cloneElement(oldElement)
      expect(React.isValidElement(newElement)).toBeTruthy()
      expect(newElement.props.style).toBe(style)
      style.display = 'none'
      expect(newElement.props.style.display).toBe('none')
    })
  })
  it('isValidElement', () => {
    let result
    result = React.isValidElement('')
    expect(result).toBeFalsy()
    result = React.isValidElement(() => <div></div>)
    expect(result).toBeFalsy()
    result = React.isValidElement(<div></div>)
    expect(result).toBeTruthy()
  })
})
