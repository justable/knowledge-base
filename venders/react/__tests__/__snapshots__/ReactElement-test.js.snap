// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactElement createElement 快照element的数据结构 1`] = `
<div
  className="container"
>
  hello world
</div>
`;
